src/stactools/core/utils/antimeridian.py

```diff
@@ -248,104 +247,31 @@ def enclose_poles(polygon: Polygon) -> Polygon:
-    # crossing latitude is in the other hemisphere.
-    #
-    # If we only have one crossing, we just guess that it's crossing on the
-    # closer pole.
-    if len(crossings) == 2:
-        crossings[0].north_pole = False
-        crossings[1].north_pole = True
-
-    # We work from the back of the list so we can use the indices.
-    crossings = sorted(crossings, key=lambda c: c.index, reverse=True)
-    for crossing in crossings:
-        coords = (
-            coords[0 : crossing.index + 1]
-            + crossing.enclosure()
-            + coords[crossing.index + 1 :]
-        )
-
-    return Polygon(coords)
-
-
-@dataclass
-class _Crossing:
-    index: int
-    latitude: float
-    positive_to_negative: bool
-    north_pole: bool
-
-    @classmethod
-    def from_points(
-        cls, index: int, start: Tuple[float, float], end: Tuple[float, float]
-    ) -> _Crossing:
-        latitude_delta = end[1] - start[1]
-        if start[0] > 0:
-            split_latitude = round(
-                start[1]
-                + (180.0 - start[0]) * latitude_delta / (end[0] + 360.0 - start[0]),
-                7,
-            )
-            return cls(
-                index=index,
-                latitude=split_latitude,
-                positive_to_negative=True,
-                north_pole=split_latitude > 0,
-            )
-        else:
-            split_latitude = round(
-                start[1]
-                + (start[0] + 180.0) * latitude_delta / (start[0] + 360.0 - end[0]),
-                7,
-            )
-            return cls(
-                index=index,
-                latitude=split_latitude,
-                positive_to_negative=False,
-                north_pole=split_latitude > 0,
-            )
 
-    def enclosure(self) -> List[Tuple[float, float]]:
-        if self.positive_to_negative:
-            longitudes = (180.0, -180.0)
-        else:
-            longitudes = (-180.0, 180.0)
-        if self.north_pole:
-            pole_latitude = 90.0
-        else:
-            pole_latitude = -90.0
-        return [
-            (longitudes[0], self.latitude),
-            (longitudes[0], pole_latitude),
-            (longitudes[1], pole_latitude),
-            (longitudes[1], self.latitude),
-        ]
+    Raises:
+        ValueError: Raised if the polygon was split. This is to keep the return
+        type the same until this function is removed.
+    """
+    warnings.warn(
+        (
+            "`stactools.core.utils.antimeridian.enclose_poles` is deprecated. "
+            "To fix polygons that enclose the poles, use the antimeridian "
+            "package (https://pypi.org/project/antimeridian/)."
+        ),
+        DeprecationWarning,
+    )
+    fixed = antimeridian.fix_polygon(polygon)
+    if fixed.geom_type == "Polygon":
+        return fixed
+    else:
+        raise ValueError("Input polygon was split and we cannot return a MultiPolygon")
```

Comment:
If this wasn't on the chopping block, I'd think maybe we could do a dirty `Union[Poly, Multi]`.

Reply:
Agreed, and that's what I do here: https://github.com/gadomski/antimeridian/blob/931b78f83515b2037be428f7fe3db41d14f546db/src/antimeridian/_implementation.py#L215-L217. But yeah, since we're deprecating it didn't seem to make sense to change the return signature.